Allow invoking callbacks with r-value refs

noelex opened the pull request:
This PR adds a new `ViewerCallbacks::request_to_invoke` callback, which allows extension modules to manipulate the window create by interactive/standard viewer on the viewer thread.

Currently performing visualization in extension modules is not supported as only a single `guik::LightViewer` is allowed in a process. `ViewerCallbacks::request_to_invoke` allows multiple extension modules to perform visualization in the main viewer window.

koide3 commented:
Thanks for the PR. But, I would like to avoid adding the callbacks for the viewers because I want to keep the viewers separable from the GLIM main library. Instead, you can directly use `guik::viewer()->invoke()`, which is thread-safe, from an extension module to extend the UI.

Meanwhile, I would like to merge the modifications for r-val refs if you could make a PR solely for them.

noelex commented:
Oh I didn't know about `guik::viewer()->invoke` before. That would be even better for my use case.

I removed the commit of the callback change and renamed the PR. Should be good to go.

koide3 commented:
Thanks!
